Solution1/WeatherApi/Infrastructure/BackgroundJobFilter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using BusinessLayer.Services.Abstract;
+using Hangfire;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Options;
+using System;
+using WeatherApi.Configuration;
+
+namespace WeatherApi.Infrastructure
+{
+    public class BackgroundJobFilter : IStartupFilter
+    {
+        private readonly IBackgroundJobClient _backgroundJobsClinent;
+        private readonly IOptionsMonitor<BackgroundJobConfiguration> _backgroundJobConfiguration;
+        private readonly IOptionsMonitor<WeatherApiConfiguration> _apiConfiguration;
+
+        public BackgroundJobFilter(IBackgroundJobClient backgroundJobsClinent, IOptionsMonitor<BackgroundJobConfiguration> backgroundJobConfiguration, IOptionsMonitor<WeatherApiConfiguration> apiConfiguration)
+        {
+            _backgroundJobsClinent = backgroundJobsClinent;
+            _backgroundJobConfiguration = backgroundJobConfiguration;
+            _apiConfiguration = apiConfiguration;
+
+        }
+        public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
+        {
+            _backgroundJobConfiguration.OnChange(appConfig => EnqueueBackgroundJob());
+            EnqueueBackgroundJob();
+            
+            return next;
+        }
+
+        private void EnqueueBackgroundJob()
+        {
+            _backgroundJobsClinent.Enqueue<IBackgroundJobService>(
+                x => x.UpdateJobs(
+                    _backgroundJobConfiguration.CurrentValue.CitiesOptions, 
+                    $"{_apiConfiguration.CurrentValue.CurrentWeatherUrl}{_apiConfiguration.CurrentValue.Key}"));
```

Comment:
I suggested to add static class or extension for combining these fields - you repeat same logic in several places, if structure of url will change you'll change it everywhere instead of one place